From: Thomas Renninger <trenn@suse.de>
Subject: EDAC i7core: Use mce socketid for better compatibility
References: fate#311968
Patch-Mainline: not yet (going to submit)

mce->socketid and cpu_data(mce->cpu).phys_proc_id are the same,
compare with mce_setup (in mce.c):
	m->cpu = m->extcpu = smp_processor_id();
        ...
	m->socketid = cpu_data(m->extcpu).phys_proc_id;

This makes it easier for example for XEN patches to hook into
MCE subsystem.

CC: JBeulich@novell.com
Signed-off-by: Thomas Renninger <trenn@suse.de>

---
 drivers/edac/i7core_edac.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: linux-2.6.32-SLE11-SP2/drivers/edac/i7core_edac.c
===================================================================
--- linux-2.6.32-SLE11-SP2.orig/drivers/edac/i7core_edac.c
+++ linux-2.6.32-SLE11-SP2/drivers/edac/i7core_edac.c
@@ -1845,7 +1845,7 @@ static int i7core_mce_check_error(void *
 
 #ifdef CONFIG_SMP
 	/* Only handle if it is the right mc controller */
-	if (cpu_data(mce->cpu).phys_proc_id != pvt->i7core_dev->socket)
+	if (mce->socketid != pvt->i7core_dev->socket)
 		return 0;
 #endif
 
